array.h

```diff
@@ -2737,7 +2737,8 @@ const_array_ref<U, Shape> reinterpret(const array<T, Shape, Alloc>& a) {
  * `new_shape`, with a base pointer offset `offset`. */
 template <class NewShape, class T, class OldShape>
 NDARRAY_HOST_DEVICE array_ref<T, NewShape> reinterpret_shape(
-    const array_ref<T, OldShape>& a, const NewShape& new_shape, index_t offset = 0) {
+    const array_ref<T, OldShape>& a, NewShape new_shape, index_t offset = 0) {
```

Comment:
Some places, `array_ref` and `shape` is passed by const ref, elsewhere, by value. What's the recommendation?

I think small things should always be passed by value in modern C++?

Reply:
I agree passing these by value is quite reasonable. That said, they aren't *that* small if the shape has a few dimensions (each dim has the same size as 3 pointers). I changed this one to be by value because the implementation needs to mutate it, so it would be making a copy anyways.

In this case it doesn't really matter, but for non-trivial types, this is a lot better (enables move construction if the caller wants that, instead of construct + copy).